backend/infrahub/core/diff/coordinator.py

```diff
@@ -45,12 +43,11 @@ async def get_diff(
             missing_time_range_diffs.append(enriched_diff)
         full_time_range_diffs = calculated_timeframe_diffs + missing_time_range_diffs
         full_time_range_diffs.sort(key=lambda dr: dr.from_time)
-        first_diff = full_time_range_diffs.pop(0)
-        combined_diff = reduce(
-            lambda first, second: self.diff_combiner.combine(earlier_diff=first, later_diff=second),
-            full_time_range_diffs,
-            first_diff,
-        )
+        combined_diff = full_time_range_diffs.pop(0)
```

Comment:
Will full_time_range_diffs always contain at least one element within this context?

Reply:
this class is going to get updated in my next PR
but you are right that this needs better handling and I've made the update in my next branch